base/abstractarray.jl

```diff
@@ -208,6 +208,13 @@ reshape(a::AbstractArray, dims::Int...) = reshape(a, dims)
 
 ## from general iterable to any array
 
+function copy!(dest, src)
+    for x in src
+        push!(dest, x)
```

Comment:
I agree that in general `append!` is repeated `push!`. So it seems the solution is to have an `f!` such that `copy!` is repeated `f!`. `f!` works for unordered collections, or collections like dicts where the "elements" contain both a key and a value, so there is no question about "where" it will be inserted.

`append!` is in fact *only* defined for Vector and BitVector, so this hasn't been probed much. We also have `union!`, which is literally implemented with repeated `push!`, and `merge!` which could easily be. These certainly overwrite existing values. So arguably we should replace `push!` with `f!` in these cases. `f!` could be, say, a 2-argument form of `insert!`.

A bit more background: in the use case in this PR, I'm either incrementally growing an Array, or building some other kind of collection. So `push!` kind of makes sense.